cmake/modules/AddSwift.cmake

```diff
@@ -548,10 +548,14 @@ function(_add_swift_runtime_link_flags target relpath_to_lib_dir bootstrapping)
       target_link_libraries(${target} PRIVATE "swiftCore")
 
       target_link_directories(${target} PRIVATE ${host_lib_dir})
+
+      # At runtime, use swiftCore in the current toolchain.
+      # FIXME: This assumes the ABI hasn't changed since the builder.
+      set(swift_runtime_rpath "$ORIGIN/${relpath_to_lib_dir}/swift/${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_LIB_SUBDIR}")
+
       if(ASRLF_BOOTSTRAPPING_MODE STREQUAL "HOSTTOOLS")
-        set(swift_runtime_rpath "${host_lib_dir}")
-      else()
-        set(swift_runtime_rpath "$ORIGIN/${relpath_to_lib_dir}/swift/${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_LIB_SUBDIR}")
+        # But before building the stdlib with the tool, use the builder libs. This should be removed in install time.
+        list(APPEND swift_runtime_rpath "${host_lib_dir}")
```

Comment:
`RUNPATH` is now like `$ORIGIN/../lib/swift/linux:/opt/swift/5.8.1/usr/lib/swift/linux`
So after the `swiftCore` is built in the build directory, that's used.